### DISCUSSION TITLE
How to set width and height for code block and output

### DISCUSSION THREAD

**Discussion by Andrzej-Andrzej** [quarto-dev/quarto-cli#5750] [Q&A]
### Description

Hi,
I would like to increase the size of code block but for only a particular slide (not for all of them) in Quarto presentation.
I want to increase width in order not to use horizontal slider bacause I have got enough space to the right (1). 
I want to scale up output as well, because it looks a bit tiny on my screen and I have got  a place for it to the right side and downwards as well (2). 

![obraz](https://github.com/quarto-dev/quarto-cli/assets/28902421/34b0b273-d640-44ae-80bc-f217109b1fae)

I included reprex file below, just copied/pasted all from my qmd file. I hope this is OK now ?
Any help would be much appreciated. Thank you very much indeed in advance.

````qmd
---
title: "Flex"
format: 
  revealjs:
    slideNumber: true
    self-contained: true
editor: visual
execute: 
  echo: true
  include: true
  output: true
---

## Slide_1

Reprex code below:

```{r}
#| label: my_chunk_1 - defining M.
#| output-location: column-fragment

library(flextable)
library(officer)
library(tidyverse)

M <- as.table(rbind(c(762, 327, 468), c(484, 239, 477)))
dimnames(M) <- list(
  gender = c("F", "M"),
  party = c("Democrat", "Independent", "Republican"))


M
```

##Slide_2

```{r}
#| label: my_chunk_2 - showing M.
#| output-location: column-fragment
M <- as.data.frame(M) 

M
```

##Slide_3

```{r}
#| label: my_chunk_3 - uncounting M.
#| output-location: column-fragment

M <- M |> uncount(Freq)

M
```

##Slide_4

```{r}
#| label: my_chunk_4 - defining f.
#| output-location: column-fragment

f <- ftable(M,
            row.vars = "gender",
            col.vars = c("party"))


ftable_to_flextable <- function( x ){

  row.vars = attr( x, "row.vars" )
  col.vars = attr( x, "col.vars" )
  rows <- rev( expand.grid( rev(row.vars), stringsAsFactors = FALSE ) )
  cols <- rev(expand.grid( rev(col.vars), stringsAsFactors = FALSE ))

  xmat <- as.matrix(x)
  cols$col_keys = dimnames(xmat)[[2]]
  xdata <- cbind(
    data.frame(rows, stringsAsFactors = FALSE),
    data.frame(xmat, stringsAsFactors = FALSE)
  )
  names(xdata) <- c(names(row.vars), cols$col_keys)

  ft <- flextable(xdata)
  ft <- set_header_df(ft, cols)
  ft <- theme_booktabs(ft)
  ft <- merge_v(ft, j = names(row.vars))
  ft
}

f
```

##Slide_5

```{r}
#| label: my_chunk_5_ftable_to_flextable
#| tbl-cap: Description of my flextable
#| output-location: column-fragment

final <- ftable(M) %>% ftable_to_flextable()

final
```

the end.
````
Here above, my code output is placed to the right side in column layout, but in default option for Quarto when output is placed down below the code, there is no space between output and code and it looks a bit squashed. How to increase distance between code and output in such a case ? As always I am grateful for your kind help, thank you.

**Comment by mcanouil** [comment]
Because you are using `engine: knitr` (implicitly), you can use `attr-*` and/or `class-*` options to add attributes/classes to the code cell you wish to modify.
Alternatively, see #4943.

**Reply by Andrzej-Andrzej** [reply]
I have read this as well:
[https://stackoverflow.com/questions/73460033/how-to-change-code-block-height-and-width-in-quarto-presentation](https://stackoverflow.com/questions/73460033/how-to-change-code-block-height-and-width-in-quarto-presentation)

Could you please if possible elaborate a bit more how to do it, ideally using my reprex if possible. What I know is that I have to prepare css file, but how to do it for only selected slides, not all of them.

**Reply by mcanouil** [reply]
See <https://bookdown.org/yihui/rmarkdown-cookbook/attr-output.html#attr-output>.
See <https://bookdown.org/yihui/rmarkdown-cookbook/hook-scroll.html#hook-scroll>.

Also you could use the R package you use to produce the table to tweak the output, such as increasing the font size, see <https://davidgohel.github.io/flextable/reference/fontsize.html>, or make sure your table fit the available width, see <https://davidgohel.github.io/flextable/reference/fit_to_width.html>.
All of this is not really Quarto, it's HTML/CSS and the R packages you use.
The same thing would apply for `rmarkdown` for instance.

**Reply by Andrzej-Andrzej** [reply]
Is it possible to do it with that three colons ":::" structure as well or that serves another purpose ?
